### ISSUE TITLE
Generate lists of PDB ids / chains for the training, validation and test sets

### ISSUE THREAD

**Issue by ostrokach** [aqlaboratory/proteinnet#1]
Hi,

I think that you have gone to great lengths to make sure that there is no information leakage between the training and the validation datasets. I would like to change my own datasets to match your training / validation split. Could you suggest a way for me to do this?

In particular, I was wondering what would be the easiest way for me to generate a list of PDB and chain ids that correspond to the training, validation, and test datasets for each of the CASP experiments?

I can see that each entry in the tensorflow files has an id, which can be something like `1WIO_1_A` , `1YX2_1_A`, `1KA1_1_A`, `1D2O_1_A`, etc. or something like `1DLC_d1dlca2`, `2BU7_d2bu7a2`, `1OMW_d1omwg-`, `1NUB_d1nubb3`, etc. Could you explain how these IDs were generated, what each part corresponds to, and what the difference is between the first type of ids (which looks like `{pdb_id}_{model_id}_{chain_id}`) and the second type of ids?

Thank you!

**Comment by alquraishi** [comment]
Thanks! Yes you should be able to use the IDs to change your own datasets, but there are some caveats.

So the easiest are the entries coming from the PDB, which as you mentioned are of the form `{pdb_id}_{model_id}_{chain_id}` and so you can use the `pdb_id` and `chain_id` fields to extract the relevant chain from the PDB file. ProteinNet is based on mmCIF files, not PDB files, and so it also uses the numeric identifier used by mmCIF files which is guaranteed to be unique. That's the `model_id` field, but you can probably get away with using old PDB files with `pdb_id` + `chain_id`.

Entries that look like `1DLC_d1dlca2` are coming from [ASTRAL](http://scop.berkeley.edu). As mentioned in the documentation ProteinNet integrates ASTRAL as well. If you want to toss out all the ASTRAL domains you can just ignore those entries, but keep in mind that some cluster members (that belong to the cluster with the ASTRAL exemplar) may be just regular PDB entries, and so you would lose them as well. For the training set if you use the 100% seq. id. subset then you don't have to worry about that, as everything is included. But for the validation set you could lose some PDB structures. You can also go back to the PDB entries and fish out the ASTRAL domains that were extracted.

Finally for the test set entries, CASP does provide a mapping to the PDB structures, e.g. [here for CASP12](http://predictioncenter.org/casp12/targetlist.cgi). You can modify the URL for other CASPs.

I will try to integrate this information in the documentation and will eventually provides the ID separately as part of the raw data.

**Comment by ostrokach** [comment]
Great, that clears things up. Thanks!
